### PR TITLE
feat: add template version creator

### PR DESCRIPTION
**Pull request by AbhineetJain**
This PR adds a created_by field to the template version table and returns the relevant data with the API calls.

## Subtasks

- [x] add a created_by field to the template version table
- [x] populate the default value of the field from the created_by field from the corresponding template
- [x] populate the field on insertion
- [x] return the field via the API

This will enable us to show the template version history (who created it and when) on the Template page for #2876.

### Request
```
http://localhost:8080/api/v2/templateversions/be409505-f1cd-40ac-8c2d-364c27469681
```
### Response
```json
{
    "id": "be409505-f1cd-40ac-8c2d-364c27469681",
    "template_id": "790457b4-bd93-43dc-9c76-cf43c7251448",
    "organization_id": "8668a031-d228-4773-abb2-bc8ac2f39716",
    "created_at": "2022-07-14T16:43:45.993399969Z",
    "updated_at": "0001-01-01T00:00:00Z",
    "name": "elastic_leavitt4",
    "job": {
        "id": "bd08dbcf-5274-49b7-bfe5-a1c4ee259cd5",
        "created_at": "2022-07-14T16:43:45.993396474Z",
        "started_at": "2022-07-14T16:43:46.172760057Z",
        "completed_at": "2022-07-14T16:43:52.543660466Z",
        "status": "succeeded",
        "worker_id": "3f74d616-df83-4557-9fd0-d31a863ed0d8",
        "storage_source": "6c621616837cca443cf54c9445c8986527e24ee12a09c3c51ffa98c3abf743be"
    },
    "readme": "---\nname: Develop code-server in Docker\ndescription: Run code-server in a Docker development environment\ntags: [local, docker]\n---\n\n# code-server in Docker\n\n## Getting started\n\nRun `coder templates init` and select this template. Follow the instructions that appear.\n\n## Supported Parameters\n\nYou can create a file containing parameters and pass the argument\n`--parameter-file` to `coder templates create`.\nSee `params.sample.yaml` for more information.\n\nThis template has the following predefined parameters:\n\n- `docker_host`: Path to (or address of) the Docker socket.\n  \u003e You can determine the correct value for this by runnning\n  \u003e `docker context ls`.\n- `docker_arch`: Architecture of the host running Docker.\n  This can be `amd64`, `arm64`, or `armv7`.\n",
    "created_by_id": "6d17ede3-bf5d-4bb9-b1c3-5dbcf8cfc13a",
    "created_by_name": "admin"
}
```

### CODE REVIEW

File: coderd/database/migrations/000030_template_version_created_by.up.sql [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+        SELECT templates.created_by FROM templates
+        WHERE template_versions.template_id = templates.id
+        LIMIT 1
+    )
+WHERE
+    template_versions.created_by IS NULL;
+
+ALTER TABLE ONLY template_versions ALTER COLUMN created_by SET NOT NULL;
+
+COMMIT;
```
Comment (AbhineetJain) [inline coderd/database/migrations/000030_template_version_created_by.up.sql:18]: @dwahler I am hoping wrapping it in `BEGIN`-`COMMIT` would help to fix the problem caused by the migration. Not really sure if this broke the server though.

Comment (AbhineetJain) [reply coderd/database/migrations/000030_template_version_created_by.up.sql:18]: Interestingly, `make gen` is failing with:
```
panic: reached retry deadline
goroutine 1 [running]:
main.main()
	/home/runner/work/coder/coder/coderd/database/dump/main.go:19 +0x80c
exit status 2
make: *** [Makefile:60: coderd/database/dump.sql] Error 1
```

Comment (AbhineetJain) [reply coderd/database/migrations/000030_template_version_created_by.up.sql:18]: Fixed the bug, the scoping logic was not good.

**Review by presleyp — APPROVED** [review]
Frontend ✅

**Comment by dwahler** [comment]
While @AbhineetJain and I were discussing this, we realized that the migration failure is probably being caused by the temporary template versions that are created by the client, in preparation for creating a new template. Those rows have `template_id` set to null (since the actual template hasn't been created yet), so we don't have any way of knowing which user created them. 

So the simplest fix is to just make `template_versions.created_by` also nullable.

**Review by dwahler — APPROVED** [review]
LGTM!

**Pull request merged**